### PR TITLE
feat(guild): add support for setting MFA level

### PR DESCRIPTION
**Pull request by suneettipirneni**
**Please describe the changes this PR makes and why it should be merged:**

- https://github.com/discord/discord-api-docs/pull/5032

Pending:
- https://github.com/discordjs/discord-api-types/pull/476

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)
- This PR includes breaking changes

### CODE REVIEW

**Review by iCrawl — APPROVED** [review]

File: packages/discord.js/src/structures/Guild.js [hunk 1]
```diff
@@ -1138,6 +1138,22 @@ class Guild extends AnonymousGuild {
     return this;
   }
 
+  /**
+   * Sets the guild's MFA level
+   * @param {GuildMFALevel} level The MFA level
+   * @param {string} reason Reason for changing the guild's MFA level
+   * @returns {Promise<Guild>}
+   */
+  async setMFALevel(level, reason) {
```
Comment (kyranet) [inline packages/discord.js/src/structures/Guild.js:1147]: Since we're going forward with #8026, we should consolidate this into a single parameter.

**Review by kyranet — CHANGES_REQUESTED** [review]

File: packages/discord.js/typings/index.d.ts [hunk 2]
```diff
@@ -1154,6 +1154,7 @@ export class Guild extends AnonymousGuild {
   public setVerificationLevel(verificationLevel: GuildVerificationLevel | null, reason?: string): Promise<Guild>;
   public setPremiumProgressBarEnabled(enabled?: boolean, reason?: string): Promise<Guild>;
   public setWidgetSettings(settings: GuildWidgetSettingsData, reason?: string): Promise<Guild>;
+  public setMFALevel(level: GuildMFALevel): Promise<Guild>;
```
Comment (SpaceEEC) [inline packages/discord.js/typings/index.d.ts:1157]: Reason is missing in the typings:
```suggestion
  public setMFALevel(level: GuildMFALevel, reason?: string): Promise<Guild>;
```

File: packages/discord.js/src/structures/Guild.js [hunk 3]
```diff
@@ -1138,6 +1138,22 @@ class Guild extends AnonymousGuild {
     return this;
   }
 
+  /**
+   * Sets the guild's MFA level
+   * @param {GuildMFALevel} level The MFA level
+   * @param {string} reason Reason for changing the guild's MFA level
```
Comment (SpaceEEC) [inline packages/discord.js/src/structures/Guild.js:1144]: The parameter should be optional:
```suggestion
   * @param {string} [reason] Reason for changing the guild's MFA level
```

**Review by SpaceEEC — CHANGES_REQUESTED** [review]

File: packages/discord.js/src/structures/Guild.js [hunk 4]
```diff
@@ -1138,6 +1138,27 @@ class Guild extends AnonymousGuild {
...
+   * @property {GuildMFALevel} level The MFA level
+   * @property {string} [reason] Reason for changing the guild's MFA level
+   */
+
+  /**
+   * Sets the guild's MFA level
+   * @param {GuildMFALevelOptions} options The MFA level
+   * @returns {Promise<Guild>}
+   */
+  async setMFALevel({ level, reason }) {
```
Comment (didinele) [inline packages/discord.js/src/structures/Guild.js:1152]:
```suggestion
  async setMFALevel(level, reason) {
```

Also update the JSDoc and remove the `GuildMFALevel` typedef above

File: packages/discord.js/typings/index.d.ts [hunk 5]
```diff
@@ -1154,6 +1154,7 @@ export class Guild extends AnonymousGuild {
   public setVerificationLevel(verificationLevel: GuildVerificationLevel | null, reason?: string): Promise<Guild>;
   public setPremiumProgressBarEnabled(enabled?: boolean, reason?: string): Promise<Guild>;
   public setWidgetSettings(settings: GuildWidgetSettingsData, reason?: string): Promise<Guild>;
+  public setMFALevel(options: GuildMFALevelOptions): Promise<Guild>;
```
Comment (didinele) [inline packages/discord.js/typings/index.d.ts:1157]:
```suggestion
  public setMFALevel(level: GuildMFALevel, reason?: string): Promise<Guild>;
```

Also get rid of the `GuildMFALevelOptions` interface.

**Review by didinele — CHANGES_REQUESTED** [review]
I know Kyra pointed out otherwise in an earlier review, but regarding #8026, we opted not updating `set*` methods as `setX({ x: value })` seems redundant.

Other than that, LGTM

**Review by iCrawl — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Pull request merged**